AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -349,17 +374,110 @@ - (void)createOrReplaceWithPartition:(NSString *)partition
+                          documentId:nil
+                          httpMethod:kMSHttpMethodGet
+                                body:nil
+                   additionalHeaders:additionalHeaders
+                   completionHandler:^(NSData *_Nullable data, NSHTTPURLResponse *_Nullable response, NSError *_Nullable cosmosDbError) {
+                     // If not OK.
+                     if (!data || [MSDataSourceError errorCodeFromError:cosmosDbError] != MSACDocumentSucceededErrorCode) {
+                       MSLogError([MSDataStore logTag], @"Not able to retrieve documents: %@", [cosmosDbError description]);
+                       MSDataSourceError *dataSourceCosmosDbError = [[MSDataSourceError alloc] initWithError:cosmosDbError];
+                       MSPaginatedDocuments *documents = [[MSPaginatedDocuments alloc] initWithError:dataSourceCosmosDbError];
+                       completionHandler(documents);
+                       return;
+                     }
+
+                     // Deserialize the list payload and try to get the array of documents.
+                     NSError *deserializeError;
+                     id jsonPayload = [NSJSONSerialization JSONObjectWithData:(NSData *)data options:0 error:&deserializeError];
+                     NSDictionary *jsonPayloadDict = nil;
+                     NSArray *jsonDocuments = nil;
+                     if (!deserializeError && jsonPayload && [(NSObject *)jsonPayload isKindOfClass:[NSDictionary class]]) {
+                       jsonPayloadDict = (NSDictionary *)jsonPayload;
+                     }
+                     if (jsonPayloadDict && [jsonPayloadDict objectForKey:@"Documents"] &&
+                         [(NSObject *)jsonPayloadDict[@"Documents"] isKindOfClass:[NSArray class]]) {
+                       jsonDocuments = jsonPayloadDict[@"Documents"];
+                     }
+                     if (!jsonDocuments) {
+                       if (!deserializeError) {
+                         deserializeError = [[NSError alloc] initWithDomain:kMSACDataStoreErrorDomain
+                                                                       code:MSACDataStoreErrorJSONSerializationFailed
+                                                                   userInfo:@{NSLocalizedDescriptionKey : @"Can't deserialize documents"}];
+                       }
+                       MSDataSourceError *dataSourceDeserializeError = [[MSDataSourceError alloc] initWithError:deserializeError];
+                       MSPaginatedDocuments *documents = [[MSPaginatedDocuments alloc] initWithError:dataSourceDeserializeError];
+                       completionHandler(documents);
+                       return;
+                     }
+
+                     // Parse the documents.
+                     NSMutableArray<MSDocumentWrapper *> *items = [NSMutableArray new];
+                     for (id document in jsonDocuments) {
+                       // Deserialize current document.
```

Comment:
FYI: The formatter doesn't seem to like new lines directly after `{` - it will often remove them.  We use the formatter as the final check so this may have to stay like this

Reply:
Confirmed, the formatter just removed the new line again...

Reply:
On a for loop? It only does this on blocks that's so weird!